lib/views/home/widgets/menu/row_item_widget.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:Medito/constants/constants.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+
+class RowItemWidget extends StatelessWidget {
+  const RowItemWidget({
+    super.key,
+    required this.title,
+    required this.iconCodePoint,
+    this.isShowUnderline = true,
+    this.onTap,
+  });
+
+  final String title;
+  final String iconCodePoint;
+  final bool isShowUnderline;
+  final void Function()? onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    var border = Border(
+      bottom: isShowUnderline
+          ? BorderSide(
+              width: 0.7,
+              color: ColorConstants.darkMoon,
+            )
+          : BorderSide.none,
+    );
+    var icon = int.parse('0x$iconCodePoint');
```

Comment:
I would much rather use a mapper class that says like: 

`if(iconStringFromBackend == 'edit') return Icons.edit_outlined` etc. 

@scouech Can you shed some light on this please? Why are we passing the icon code?

Reply:
yeah codepoint :)

Reply:
so that we dont have to translate what the icon name is as a codepoint. We just send a codepoint from the backend. 

However, is there a way @osamaasifoak that if the codepoint is not valid to display a default one? wha would happen if the value is like 0000000 as a codepoint for example?

Reply:
Good point @michaelcspeed, Previously, I was showing the icons using the if else statement. But, then we decided to add the codePoints in the API. Because Flutter is also showing the icons using the codePoints.
 e.g. `static const IconData ten_k = IconData(0xe000, fontFamily: 'MaterialIcons');`
 
 Yes @scouech, I will see how to show the default icon.

Reply:
@scouech, just checked we don't have any error callback in IconData or the Icon widget. So the only drawback here is that the PointCodes must be valid. 

@michaelcspeed  do you have any suggestions for us?

Reply:
So what happens if you set an invalid code point? No exception is thrown? Is the widget just empty?

Reply:
No exception and yeah just an empty widget

Reply:
I think that's fine, as long as the app doesn't crash!